src/OpenAI/Model/Gpt.php

```diff
@@ -11,22 +11,31 @@
 use PhpLlm\LlmChain\Response\Choice;
 use PhpLlm\LlmChain\Response\Response;
 use PhpLlm\LlmChain\Response\ToolCall;
+use Webmozart\Assert\Assert;
 
 final class Gpt implements LanguageModel
 {
     public function __construct(
         private readonly Runtime $runtime,
         private ?Version $version = null,
         private readonly float $temperature = 1.0,
+        private readonly float $topP = 1.0,
```

Comment:
What do you think in general of having those parameters here? Struggle a bit with my initial decision - should maybe only the part of the `$options` of `call(..)` method?

Reply:
I think for default settings for fine tuning this position is fitting. But maybe all settings available should be put to a config object? 

When i have a look to the OpenAI Playground ... there are more settings beside temperature and topP for every request. So this would stack up to this amount.

![image](https://github.com/user-attachments/assets/ada13d26-a45c-4fdb-8330-15c0d07e86c4)

 Maybe this could also be utilized as argument by the user?

Reply:
maintaining a config object would be hard - the api contract is growing and depends on versions and platform. we could tackle that of course with having designated extension points, but having an array is way more flexible for now :speak_no_evil: :laughing:

Reply:
We can go with `$options` array then, but we couldn't validate them, just pass them to the model -> fire and forget... 🤷‍♂️

Reply:
We need way better error handling tho for http calls. Was quite lazy with that ... 👼